app/models/miq_server.rb

```diff
@@ -38,6 +38,7 @@ class MiqServer < ApplicationRecord
   virtual_delegate :description, :to => :zone, :prefix => true, :allow_nil => true, :type => :string
 
   validate :validate_zone_not_maintenance?
+  validate :zone_unchcanged_in_pods, :on => :update
```

Comment:
This successfully rejects the change in zone, but the UI is not displaying an error. I'll try to fix that over there.